frontend/legacyToolCollectionRoutes.js

```diff
@@ -0,0 +1,122 @@
+function getToolRedirects() {
+   const toolLegacyRoutes = [
+      'AC_Inlets',
+      'Accessories',
+      'Adapters',
+      'Adhesive_Pads',
+      'Adhesive_Strips',
+      'Antennas',
+      'Audio',
+      'Batteries',
+      'Bearings',
+      'Belts',
+      'Bluetooth_Boards',
+      'Bumpers',
+      'Buttons',
+      'Cables',
+      'Cameras',
+      'Card_Cages',
+      'Case_Components',
+      'Charger_Boards',
+      'Clearance',
+      'Click_Wheels',
+      'Clips',
+      'Computers',
+      'Controllers',
+      'DC-In_Boards',
+      'DC-to-DC_Boards',
+      'Daughterboards',
+      'Digitizers',
+      'Display_Components',
+      'Displays',
+      'Dock_Connectors',
+      'Docks',
+      'Earbuds',
+      'Fans',
+      'Filters',
+      'Flash',
+      'Front_Panels',
+      'Gaskets',
+      'Graphics_Cards',
+      'Hard_Drive_Brackets',
+      'Hard_Drive_Enclosures',
+      'Hard_Drive_Kits',
+      'Hard_Drives',
+      'Hard_Drives_(PATA)',
+      'Hard_Drives_(SATA)',
+      'Headphone_Jacks',
+      'Heat_Sinks',
+      'Hinges',
+      'I/O_Board',
+      'Ice_Maker',
+      'Induction_Coil',
+      'Inverters',
+      'Joysticks',
+      'Keyboards',
+      'Kits',
+      'LCDs',
+      'Lamp',
+      'Lasers',
+      'Lenses',
+      'Lightning_Connector',
+      'Logic_Boards',
+      'MagSafe_Boards',
+      'Magnets',
+      'Manuals',
+      'Memory_Maxxer_Kits',
+      'Microphones',
+      'Microsoft_ASP',
+      'Midframe',
+      'Modems',
+      'Motherboards',
+      'Motors',
+      'Optical_Drives',
+      'Organization',
+      'PCIe',
+      'PC_Cards',
+      'Ports',
+      'Power_Jacks',
+      'Power_Supplies',
+      'Processors',
+      'Propellers',
+      'Pulleys',
+      'Pumps',
+      'RAM',
+      'RJ-11_Boards',
+      'Retail_Display',
+      'Rubber_Feet',
+      'SD_Card',
+      'SD_Card_Slots',
+      'SIM',
+      'SSD_Enclosures',
+      'SSD_Upgrade_Kits',
+      'SSDs',
+      'Screen_Protectors',
+      'Screens',
+      'Screwdrivers_and_Nutdrivers',
+      'Screws',
+      'Sensors',
+      'Software',
+      'Speakers',
+      'Straps',
+      'Styluses',
+      'Switches',
+      'Teardown_Case',
+      'Test_Cables',
+      'Textile_Repair',
+      'Thermal_Pads',
+      'Timing_Control_Boards',
+      'Trackpads',
+      'USB_Boards',
+      'Vaude',
+      'Vibrators',
+      'Wi-Fi_Boards',
+      'Wireless',
+   ];
+
+   return toolLegacyRoutes
+      .map((route) => route.replaceAll(/([\/\(\)])/g, '\\$&'))
+      .join('|');
```

Comment:
This regex generating regex is a doozy

Reply:
its not really that bad its basically find possibly regex breaking characters (ie `(`  `)` `/`) and escape them with an escaped backslash.